src/NClient/NClient.Core/Helpers/TypeExtensions.cs

```diff
@@ -61,6 +61,61 @@ public static MethodInfo[] GetInterfaceMethods(this Type type, bool inherit = fa
             return methodInfos.ToArray();
         }
 
+        public static MethodInfo[] GetUnhiddenInterfaceMethods(this Type type, bool inherit = false)
```

Comment:
I think it's worth writing a test for this method, because the logic is not simple :)